CoreStubs/ArchiveControl/module.php

```diff
@@ -52,7 +52,39 @@ public function ChangeVariableID(int $OldVariableID, int $NewVariableID)
 
     public function DeleteVariableData(int $VariableID, int $StartTime, int $EndTime)
     {
-        throw new Exception("'DeleteVariableData' is not yet implemented");
+        $archivedData = $this->GetVariableData($VariableID);
+        $Values = $archivedData['Values'];
+
+        if ($StartTime === 0 && $EndTime === 0) {
+            $this->SetLoggingStatus($VariableID, false);
+        }
+
+        //Get the first and last timestamp if it is necessary
+        if (array_key_exists('TimeStamp', $Values)) {
+            if ($EndTime === 0 || $EndTime < end($Values)['TimeStamp']) {
+                $EndTime = end($Values)['TimeStamp'];
+            }
+            if ($StartTime === 0 || $StartTime > reset($Values)['TimeStamp']) {
+                $StartTime = reset($Values)['TimeStamp'];
+            }
+        }
+
+        //Get the Start and endkey and delete the values between them
+        $timeStampKeys = array_column($Values, 'TimeStamp');
+        $startKey = array_search($StartTime, $timeStampKeys);
+        $endKey = array_search($EndTime, $timeStampKeys);
+
+        if ($startKey !== false && $endKey !== false) {
```

Comment:
Die !==-Blöcke gerne in Klammern. Das ist so auch korrekt, Klammern verbessern da aber die Lesbarkeit

Ich bin nicht sicher ob Ungleichheit für beide Werte hier immer gegeben ist falls etwas gelöscht werden muss. Aber da musst du mal schauen, wenn du die Bestimmung von $startKey und $endKey anpasst, siehe oberer Kommentar